src/foam/dao/HTTPSink.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao;
+
+import foam.core.Detachable;
+import foam.core.FObject;
+import foam.lib.Outputter;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpException;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+public class HTTPSink
+    extends AbstractSink
+{
+  protected String url_;
+  protected Outputter outputter_;
```

Comment:
If two threads hit the same DUG at the same point their output will be interlaced. Just create a new Outputter for each put().